packages/contracts-bedrock/src/L1/BatchInbox.sol

```diff
@@ -0,0 +1,75 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.28;
+
+import {NitroValidator} from "@nitro-validator/NitroValidator.sol";
+import {CborDecode} from "@nitro-validator/CborDecode.sol";
+import {CertManager} from "@nitro-validator/CertManager.sol";
+import {ECDSA} from "@openzeppelin/contracts/utils/cryptography/ECDSA.sol";
+import {OwnableUpgradeable} from "@openzeppelin/contracts-upgradeable/access/OwnableUpgradeable.sol";
+import {ISemver} from "interfaces/universal/ISemver.sol";
+
+contract BatchInbox is OwnableUpgradeable, ISemver, NitroValidator {
+    /// @notice Semantic version.
+    /// @custom:semver 1.0.1
+    string public constant version = "1.0.1";
+
+    using CborDecode for bytes;
+
+    uint256 public constant MAX_AGE = 60 minutes;
+
+    /// @notice Mapping of valid PCR0 values for batchers
+    mapping(bytes32 => bool) public validPCR0s;
+
+    // @notice Mapping of attested batchers, effectively used as a hashset, i.e. the value is unused
+    mapping(address => bool) public attestedBatchers;
+
+    constructor(
+        CertManager certManager
+    ) NitroValidator(certManager) OwnableUpgradeable() {
+        initialize();
+    }
+
+    function initialize() public initializer {
+        __Ownable_init();
+        attestedBatchers[owner()] = true;
+    }
+
+    function registerPCR0(bytes calldata pcr0) external onlyOwner {
+        validPCR0s[keccak256(pcr0)] = true;
+    }
+
+    function submitBatch(
+        bytes calldata commitment,
+        bytes calldata _signature
+    ) external {
+        // https://github.com/ethereum/go-ethereum/issues/19751#issuecomment-504900739
+        bytes memory signature = _signature;
```

Comment:
Why are we doing this? You can call the recover like this: https://github.com/base/op-enclave/blob/8960aa87053bdd954ea356a340359572e9633fd4/contracts/src/OutputOracle.sol#L128-L133